src/main/java/org/broadinstitute/hellbender/engine/dataflow/DataflowCommandLineProgram.java

```diff
@@ -55,12 +64,31 @@ protected Object doWork() {
         options.setStagingLocation(stagingLocation);
         options.setRunner(this.runner.runner);
         Pipeline p = Pipeline.create(options);
+        DataflowWorkarounds.registerGenomicsCoders(p);
         setupPipeline(p);
-        p.run();
+        runPipeline(p);
         return null;
     }
 
     /**
+     * Runs a {@link Pipeline} and unwraps RuntimeExceptions caused by UserExceptions back into UserExceptions
+     */
+    @VisibleForTesting
+    protected static void runPipeline(Pipeline p) {
+        try{
+            p.run();
+        } catch( RuntimeException e  ){
+            //Data flow catches our UserExceptions and wraps them in RuntimeException
```

Comment:
I like this comment! A clear explanation for what otherwise would have been a mysterious piece of code